src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -736,36 +790,81 @@ public void ExecuteOperationBlockActions(
             bool isGeneratedCode,
             CancellationToken cancellationToken)
         {
-            ExecuteBlockActionsCore<OperationBlockStartAnalyzerAction, OperationBlockAnalyzerAction, OperationAnalyzerAction, IOperation, int>(
-                operationBlockStartActions, operationBlockActions, operationBlockEndActions, analyzer,
-                declaredNode, declaredSymbol, operationBlocks, (blocks) => operations, semanticModel,
-                getKind: null, filterSpan, isGeneratedCode, cancellationToken);
+            Debug.Assert(!operationBlocks.IsEmpty);
+
+            // The actions we discover in 'addActions' and then execute in 'executeActions'.
+            var ephemeralActions = ArrayBuilder<OperationAnalyzerAction>.GetInstance();
+            ExecuteBlockActionsCore(
+                startActions,
+                actions,
+                endActions,
+                analyzer,
+                declaredNode,
+                declaredSymbol,
+                operationBlocks,
+                semanticModel,
+                filterSpan,
+                isGeneratedCode,
+                addActions: static (startAction, endActions, args, cancellationToken) =>
+                {
+                    var (@this, startActions, analyzer, declaredNode, operationBlocks, declaredSymbol, operations, semanticModel, filterSpan, isGeneratedCode, ephemeralActions) = args;
+                    var scope = new HostOperationBlockStartAnalysisScope(startAction.Analyzer);
+                    var startContext = new AnalyzerOperationBlockStartAnalysisContext(
+                        scope, operationBlocks, declaredSymbol, semanticModel.Compilation, @this.AnalyzerOptions,
+                        @this.GetControlFlowGraph, declaredNode.SyntaxTree, filterSpan, isGeneratedCode, cancellationToken);
+
+                    // Catch Exception from the start action.
+                    @this.ExecuteAndCatchIfThrows(
+                        startAction.Analyzer,
+                        static args => args.startAction.Action(args.startContext),
+                        argument: (startAction, startContext),
+                        new AnalysisContextInfo(@this.Compilation, declaredSymbol),
+                        cancellationToken);
+
+                    endActions.AddAll(scope.OperationBlockEndActions);
+                    ephemeralActions.AddRange(scope.OperationActions);
```

Comment:
same cleanup here.  This also makes the lambda/arguments pattern used here consistent with every other call out to user code in the rest of the file, which is very nice for simplicity and consistency.   You wll see that very clearly in the next pr.

Reply:
this comment no longer applies as this refactoring was made in a prior PR that already went in.